tools/topology/topology2/platform/intel/i2s-dumb-amp-echo-ref.conf

```diff
@@ -0,0 +1,73 @@
+#
+# Pcm, Dai instance & route index all start from 200 to avoid collisions
+#
+
+Define {
+	ECHO_REF_STREAM_NAME	"Echo reference"
+}
+
+Object.Dai {
+	# override amp instance's i2s config for echo reference
+	SSP."200" {
+		direction	"duplex"
+		quirks		"lbm_mode"
+	}
+}
+
+Object.Pipeline {
+	passthrough-capture.2 {
+		index	$ECHO_REF_HOST_PIPELINE_ID
+
+		Object.Widget.copier.1 {
+			stream_name	$ECHO_REF_STREAM_NAME
+		}
+	}
+
+	passthrough-copier-be.2 {
+		index	$ECHO_REF_DAI_PIPELINE_ID
+
+		Object.Widget.copier."1" {
+			dai_index	1
+			dai_type	"SSP"
+			type		dai_out
+			copier_type	"SSP"
+			stream_name	$SPEAKER_CODEC_NAME
+			node_type	$I2S_LINK_INPUT_CLASS
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+		}
+	}
+}
+
+Object.PCM {
+	pcm."200" {
+		name	"EchoRef"
+		id	27
+		direction	"capture"
+		Object.Base.fe_dai.1 {
+			name	"EchoRef"
```

Comment:
one feature of tplg2 that is not so widely known is when there's an object embedded within another and there's a shared attribute between them, like name in this case,  and they have the same value, there's no need to repeat it inside the child object. So you can just simple have Object.Base.fe_dai.1 {} and it will work fine.

Reply:
Ack, it's a good learning.